src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
I'm not sure.

`ExecuteTrajectory` only returns the trajectory in case it is modified internally (e.g. simplified, smoothed, or re-timed). In my refactor, this requires one or more trips through the planning pipeline and could return a future trajectory.

On the other hand, it would be more useful to return a `future` that tracks the state of the trajectory being executed. This would eliminate the need for `WaitForController` and the tri-state `timeout` named parameter to `ExecuteTrajectory`. It would also, finally, give us a good method of querying error codes from the controller (e.g. stalled, aborted, etc).

Thoughts?